core/services/synchronization/stats_pusher.go

```diff
@@ -119,14 +124,19 @@ func (sp *statsPusher) Close() error {
 	if sp.cancel != nil {
 		sp.cancel()
 	}
+
 	gormCallbacksMutex.Lock()
-	_ = sp.ORM.RawDB(func(db *gorm.DB) error {
+	err := sp.ORM.RawDB(func(db *gorm.DB) error {
 		db.Callback().Create().Remove(createCallbackName)
 		db.Callback().Update().Remove(updateCallbackName)
 		return nil
 	})
+	if err != nil {
```

Comment:
Same here, we never unlock the mutex.

Reply:
Good catch, since this is old code I'm going to address in a separate PR: https://www.pivotaltracker.com/story/show/175381537

Reply:
We unlock the mutex on line 37, this if block only logs.